packages/admin/cms-admin/src/preview/block/BlockPreview.tsx

```diff
@@ -28,6 +29,15 @@ function BlockPreview({
         }
     }, [iFrameBridge, previewState]);
 
+    const previewUrl = new URL(url);
+    const params = {
+        host: previewUrl.host,
+        path: previewUrl.pathname,
+    };
+    const sitesConfigContext = useSitesConfig();
+    const baseUrl = sitesConfigContext.sitePreviewUrl ?? `${previewUrl.protocol}://${previewUrl.host}`;
+    const initialPageUrl = `${sitesConfigContext.sitePreviewUrl ?? baseUrl}/api/block-preview?previewParams=${window.btoa(JSON.stringify(params))}`;
```

Comment:
```suggestion
    const initialPageUrl = `${baseUrl}/api/block-preview?previewParams=${window.btoa(JSON.stringify(params))}`;
    
defaults to siterPreviewUrl already above
```

Reply:
Why btoa and not encodeURIComponent?

Reply:
Why do you need to pass host at all? And why don't you use a simpler ?path=... instead of the JSON?